### DISCUSSION TITLE
What's the process of ArgoCD pull for Github repository changes?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [argoproj/argo-cd#17629] [Q&A] [answered]
Hello everyone,
I hope your day is going well. I'm curious to learn more details about the ArgoCD pull process for changes in the repo. I've seen documentation mentioning that it happens every 3 minutes, but there's no coverage of what exactly happens during a pull. In particular:
* Would there be a full repository checkout done by some Argo component?
* Is there some caching of the repository state, so that only new updates are pulled?
* Does it work only with deltas or otherwise partial repository state to get most up-to-date info?

I'm trying to understand if it's okay to auto sync from a large monorepo or if I should use a separate smaller repo. I've heard that process may be slow for large repos, but don't know the details.
Thank you!

**Comment by davidfalconego** [comment, marked as answer]
![operacion](https://github.com/argoproj/argo-cd/assets/18756337/cf8af845-23e4-4c2f-b346-8c842787006c)
Take a look
